packages/components/src/unit-control/types.ts

```diff
@@ -24,3 +37,43 @@ export type WPUnitControlUnit = {
 };
 
 export type WPUnitControlUnitList = Array< WPUnitControlUnit > | false;
+
+export interface UnitControlOnChangeCallback {
```

Comment:
It looks like this one needed to be an interface so that it can be used with a function. Let me know if that's not correct, though!

Reply:
Have you tried writing it as

```ts
export type UnitControlOnChangeCallback = (
	nextValue: string | undefined,
	extra: {
		event: SyntheticEvent< HTMLSelectElement | HTMLInputElement >;
		data?: WPUnitControlUnit;
	}
) => void;
```

?

Reply:
After reading more fo your comments, I noticed the similarity between this type and the `onChange` prop on the `InputFieldProps` type (defined for the `InputControl` component).

I played around a little bit, trying to come up with something DRY-er, but I'm not sure if this would look better:

<details>
<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/input-control/types.ts b/packages/components/src/input-control/types.ts
index 971075e56a..71be180a3a 100644
--- a/packages/components/src/input-control/types.ts
+++ b/packages/components/src/input-control/types.ts
@@ -31,15 +31,17 @@ interface BaseProps {
 	size?: 'default' | 'small';
 }
 
+export type InputChangeEvent< E = ChangeEvent< HTMLInputElement >, P = {} > = (
+	nextValue: string | undefined,
+	extra: { event: E } & P
+) => void;
+
 export interface InputFieldProps extends BaseProps {
 	dragDirection?: DragDirection;
 	dragThreshold?: number;
 	isDragEnabled?: boolean;
 	isPressEnterToChange?: boolean;
-	onChange?: (
-		nextValue: string | undefined,
-		extra: { event: ChangeEvent< HTMLInputElement > }
-	) => void;
+	onChange?: InputChangeEvent;
 	onValidate?: (
 		nextValue: string,
 		event?: SyntheticEvent< HTMLInputElement >
diff --git a/packages/components/src/unit-control/types.ts b/packages/components/src/unit-control/types.ts
index d843266fc9..6ad0bfad72 100644
--- a/packages/components/src/unit-control/types.ts
+++ b/packages/components/src/unit-control/types.ts
@@ -8,6 +8,7 @@ import type { CSSProperties, SyntheticEvent } from 'react';
  * Internal dependencies
  */
 import type { StateReducer } from '../input-control/reducer/state';
+import type { InputChangeEvent } from '../input-control/types';
 
 export type Value = number | string;
 
@@ -38,15 +39,10 @@ export type WPUnitControlUnit = {
 
 export type WPUnitControlUnitList = Array< WPUnitControlUnit > | false;
 
-export interface UnitControlOnChangeCallback {
-	(
-		nextValue: string | undefined,
-		extra: {
-			event: SyntheticEvent< HTMLSelectElement | HTMLInputElement >;
-			data?: WPUnitControlUnit;
-		}
-	): void;
-}
+export type UnitControlOnChangeCallback = InputChangeEvent<
+	SyntheticEvent< HTMLSelectElement | HTMLInputElement >,
+	{ data?: WPUnitControlUnit }
+>;
 
 export type UnitSelectControlProps = {
 	className?: string;

```
</details>